financial-services-accelerator/consent-mgt-api/org.wso2.financial.services.accelerator.consent.mgt.api.service/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/api/service/impl/ConsentCoreServiceImpl.java

```diff
@@ -0,0 +1,852 @@
+            try {
+                // Retrieve the detailed consent resource
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Retrieving detailed consent for consent ID: %s",
+                            consentId.replaceAll("[\r\n]", "")));
+                }
+                DetailedConsentResource retrievedDetailedConsentResource = consentCoreDAO
+                        .getDetailedConsentResource(connection, consentId, orgId);
+
+                // Commit transactions
+                DatabaseUtils.commitTransaction(connection);
+                log.debug(ConsentCoreServiceConstants.TRANSACTION_COMMITTED_LOG_MSG);
+                return retrievedDetailedConsentResource;
+            } catch (ConsentDataRetrievalException e) {
+                log.error(ConsentCoreServiceConstants.DATA_RETRIEVE_ERROR_MSG, e);
+                throw new ConsentMgtException(e.getConsentError());
+            }
+        } catch (SQLException e) {
+            log.error(ConsentError.DETAILED_CONSENT_RETRIEVAL_ERROR.getMessage().replaceAll("[\r\n]", ""), e);
+            throw new ConsentMgtException(ConsentError.DETAILED_CONSENT_RETRIEVAL_ERROR);
+        }
+    }
+
+    @Override
+    public ArrayList<DetailedConsentResource> searchDetailedConsents(String orgId, ArrayList<String> consentIDs,
+                                                                     ArrayList<String> clientIDs,
+                                                                     ArrayList<String> consentTypes,
+                                                                     ArrayList<String> consentStatuses,
+                                                                     ArrayList<String> userIDs, Long fromTime,
+                                                                     Long toTime,
+                                                                     Integer limit, Integer offset) throws
+            ConsentMgtException {
+
+        // Input parameters except limit and offset are not validated since they are validated in the DAO method
+        ArrayList<DetailedConsentResource> detailedConsentResources;
+
+        try (Connection connection = DatabaseUtils.getDBConnection()) {
+            try {
+                ConsentCoreDAO consentCoreDAO = ConsentStoreInitializer.getInitializedConsentCoreDAOImpl();
+
+                log.debug("Searching detailed consents");
+                detailedConsentResources = consentCoreDAO.searchConsents(connection, orgId, consentIDs, clientIDs,
+                        consentTypes, consentStatuses, userIDs, fromTime, toTime, limit, offset);
+
+            } catch (ConsentDataRetrievalException e) {
+                log.error(ConsentCoreServiceConstants.DETAIL_CONSENT_SEARCH_ERROR_MSG, e);
+                throw new ConsentMgtException(ConsentError.CONSENT_SEARCH_ERROR);
+            }
+            // Commit transactions
+            DatabaseUtils.commitTransaction(connection);
```

Comment:
Why we're committing the transaction for a search query? Please check other places as well